### PR TITLE
ci(release): add dispatch GHA for snapcraft classic release

### LINKED ISSUE #1079: Using ORAS from snap can't authenticate with gcloud credential helper
### What happened in your environment?

I'm using oras like this:
`oras push us-central1-docker.pkg.dev/... file.tar.gz`

I have docker configured to use gcloud auth using this command:
`gcloud auth configure-docker us-central1-docker.pkg.dev`

which sets up the ~/.docker/config.json like so:
```
{
  "credHelpers": {
    "us-central1-docker.pkg.dev": "gcloud"
  }
}
```

If I use oras from snap, authentication fails, but if I use the downloaded oras binary it works fine

### What did you expect to happen?

Authentication doesn't depend on 

### How can we reproduce it?

Since this requires authenticating with google cloud platform, replicating my setup isn't the easiest. I can help someone setup a similar environment if necessary 

### What is the version of your ORAS CLI?

1.0.0, which is the latest version in snap

### What is your OS environment?

Ubuntu 20.04

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR adds a Github Action which can be run manually to for snapcraft releasing.
image

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1079

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by qweeah** [comment]
Making this PR draft since the login credential has not been added to repo secrets. (see [instruction](https://github.com/snapcore/action-publish?tab=readme-ov-file#store-login))

**Comment by SamirPS** [comment]
@qweeah Do you need help for this PR ?

**Comment by qweeah** [comment]
> @qweeah Do you need help for this PR ?

Thanks @SamirPS, currently I don't need any help. I have walked through the build and publish part on Snapcraft side, just need to tune the release process to fit ORAS' release practice.

**Comment by qweeah** [comment]
After this being merged, we still need to run below command to add the content of `exported.txt` to oras's Actions secret named `STORE_LOGIN`

```bash
snapcraft export-login --snaps=oras \
      --acls package_access,package_push,package_update,package_release \
      exported.txt
```

File: .github/workflows/release-snap.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      isStable:
+        type: boolean
+        description: 'check for stable release'
+        default: false
+
+jobs:
+  release-snap:
+    strategy:
+      matrix:
+        arch: [amd64,arm64,s390x]
```
Comment (shizhMSFT) [inline .github/workflows/release-snap.yml:31]: This this array of string require quotes?

File: .github/workflows/release-snap.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        run: |
+          sed -i 's/{VERSION}/${{ steps.version.outputs.version }}/g' snapcraft.yaml
+          sed -i 's/{ARCH}/${{ matrix.arch }}/g' snapcraft.yaml
+          cat snapcraft.yaml
+      - uses: snapcore/action-build@v1
+        id: build
+      - uses: snapcore/action-publish@v1
+        name: publish
+        env:
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
```
Comment (shizhMSFT) [inline .github/workflows/release-snap.yml:57]: What's `STORE_LOGIN`?

Comment (qweeah) [reply .github/workflows/release-snap.yml:57]: It's an action secret that needs to be added to Github Actions Secrets, can be generated by the owner snapcraft account login. E.g. below command generates the secret into `exported.txt`:

```bash
snapcraft export-login --snaps=oras \
      --acls package_access,package_push,package_update,package_release \
      exported.txt
```

File: .github/workflows/release-snap.yml (hunk 1 above)
Comment (qweeah) [reply .github/workflows/release-snap.yml]: Nice catch, updated

File: .github/workflows/release-snap.yml (hunk 2 above)
Comment (qweeah) [reply .github/workflows/release-snap.yml]: renamed

File: .github/workflows/release-snap.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            echo "release=stable" >> $GITHUB_OUTPUT
+          else
+            echo "release=candidate" >> $GITHUB_OUTPUT
+          fi
+          echo "version=${{ github.event.inputs.version }}" >> $GITHUB_OUTPUT
+      - name: make snapcraft
+        run: |
+          sed -i 's/{VERSION}/${{ steps.version.outputs.version }}/g' snapcraft.yaml
+          sed -i 's/{ARCH}/${{ matrix.arch }}/g' snapcraft.yaml
+          cat snapcraft.yaml
```
Comment (SamirPS) [inline .github/workflows/release-snap.yml:54]: do we really want to cat the file?

**Review by SamirPS — CHANGES_REQUESTED** [review]

File: .github/workflows/release-snap.yml (hunk 3 above)
Comment (qweeah) [reply .github/workflows/release-snap.yml:54]: yes, worth logging for trouble shooting.

File: .github/workflows/release-snap.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: extract version
+        id: version
+        run: |
+          if [[ "${{ github.event.inputs.version }}" == "true" ]]; then
```
[the remark this answers is not in the document]
Comment (qweeah) [reply .github/workflows/release-snap.yml:44]: ah, yes. will change to isStable

[the remark this answers is not in the document]
Comment (qweeah) [reply .github/workflows/release-snap.yml]: Just confirm in my own snap release that both stable and unstable release works

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**